### DISCUSSION TITLE
Suggested additions

### DISCUSSION THREAD

**Discussion by T-C123** [jgyates/genmon#1224] [Ideas]
Thinking about some features that might be worth adding to Genmon.

1. Would it be possible to add the ability to reboot the Raspberry Pi using Genmon's email command feature? My Pi is buried deep withing the generator enclosure with no monitor or keyboard so I added an easy to get at, power switch in case I need to force a reboot. I know it would be better to cause a reboot using the Pi's normal shut down procedure and am hoping that can be initialized with an email command.

2. Annually my generator prompts me to check the battery and I give it a visual inspection as well as a load test. Most users will not have the tool to load test a battery so they will not really know its condition. Would it be possible to record and graph the battery voltage while the engine is cranking? Over time, that would give a good indication of the battery condition and if it is beginning to weaken the cranking voltage would be expected to drop. It would not be exact due to temperature changes but could help.

Thanks for all you do to keep Genmon working!

Tom

**Comment by jgyates** [comment]
There is a reboot button on the Advanced Status page. I know that the email function would be useful to you today considering your [other thread](https://github.com/jgyates/genmon/discussions/1223#discussioncomment-11654526), but the email commands require the user to take extra cautions when using this feature, like setting up email filters to prevent third parties from sending email commands. For that reason I am hesitant to add too much functionality in the email common as it may open up some users to an attack.

I am not sure how useful that would be as reading the battery voltage my not be updated over modbus with the precision that would be useful. I graph things like battery voltage with Grafana, Influx DB and MQTT.

For reference, below is a graph of my battery voltage (using MQTT, influx DB and Grafana) over the last 7 days doing continuous readings. The last 7 days includes one charging cycle and one exercise. No dip in cranking for the battery. 

<img width="844" alt="Screenshot 2024-12-23 at 8 25 51 PM" src="https://github.com/user-attachments/assets/31386169-9136-452e-9cb6-5415af70f481" />


The Evolution controller should generate an alarm if the battery is still low after a charging cycle.

**Reply by T-C123** [reply]
I am thankful for your attention to security! With my limited expertise I don't see much of a problem if a hacker forced a reboot on the Pi although forcing repeated reboots would prevent the system from operating. Maybe during the reboot process a hacker could gain further control of the system? I defer to your experience and expertise.

I looked at your graph and don't see the expected drop in battery voltage when cranking. Maybe the modbus doesn't update fast enough to capture battery voltage during the short cranking cycle or your graph doesn't show enough detail for that short time. My battery load tester puts a load on the battery and displays the loaded voltage. I thought the load of cranking would give an indication of the battery condition.

Thanks for your consideration!

**Reply by DrTron** [reply]
> I am thankful for your attention to security!

I agree that it would be a serious security breach to allow system command like a reboot to be issued via e-mail.
But what keeps you from logging into your Pi via SSH and issuing the reboot command? As the pi user, who has sudo rights, a "sudo reboot" will suffice.

**Reply by T-C123** [reply]
The reboot question was created by my problems discussed in another thread:

https://github.com/jgyates/genmon/discussions/1223#discussioncomment-11654526

I had no ability to SSH to the Pi and was looking for options to restart the system without disassembling my generator in sub-freezing weather.

**Reply by AlexOu812** [reply]
Hi,  I don't see a reboot option on the Status page.  Where is the Advanced Status page?  I'm using V1.19.07

**Reply by jgyates** [reply]
sorry, it's on the Advanced Setting page (there is no advanced status page). To access the Advanced Settings page, double click the icon in the upper right of the web interface.

**Comment by T-C123** [comment]
another idea: When Genmon starts I get an email notification that includes the IP address. Would it be possible to include the IP address in status update emails? That would help in sorting out communication problems like I had.

Thanks always!

**Reply by jgyates** [reply]
I can add that to the list. 

One thing that you can do in the interim: the advanced status page has a setting named "User URL". You can put the full URL of your genmon landing page here. Since the pi will broadcast it's address and host name, most routers will allow you to use local DNS to resovle you IP. For example if your pi host name is "genmon" then you can ping genmon.local. In the "User URL" setting, the you can put something like http://genmon.local:8000. If you set this value then the URL provide will be added to outbound emails for alarms and status changes. on my system I use a static IP address and put http://192.168.1.10:8000 for the "User URL". Of course if you change your static IP you would need to change the User URL. With this setting, you can just click on the link in the email to pull up genmon.

**Reply by jgyates** [reply]
I check in an update that adds the IP address to the bottom of each notification email. Just update from the About page and you should be set.

**Reply by skipfire** [reply]
The emails (excluding start) have the User URL at the top, and now I see the IP at the bottom of the emails.

**Reply by T-C123** [reply]
Great! Thank you!
